### ISSUE TITLE
Reorganise tests in the project

### ISSUE THREAD

**Issue by radomird** [fastify/fastify-jwt#187]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the issue has not already been raised

### Issue

As proposed in PR #184 the tests should be moved to their own folder. Quoting the request:
> - We should take the opportunity offered by this PR to rework the file structure of the plugin and move jwt.test-d.ts to test/types/jwt.test-d.ts, rename test/test.test.js to test/jwt.test.js and update the corresponding references in package.json file.

The request was out of scope for the mentioned PR so I've moved it to a new issue.

**Comment by mcollina** [comment]
Go for it once #184 lands

**Comment by ja-bravo** [comment]
I'll take this one, will send a PR in a bit

**Comment by darkgl0w** [comment]
> I'll take this one, will send a PR in a bit

@ja-bravo done already.
